### PR TITLE
PXF: Validation for server, user-mapping, and table options

### PR DESCRIPTION
**Pull request by oliverralbertini**
Add validation for SERVER options

Co-authored-by: Oliver Albertini <oalbertini@pivotal.io>
Co-authored-by: Francisco Guerrero <aguerrero@pivotal.io>

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: contrib/pxf_fdw/pxf_option.c [hunk 1]
```diff
@@ -32,22 +36,43 @@ PG_FUNCTION_INFO_V1(pxf_fdw_validator);
...
+		{
+			resource = defGetString(def);
+
+			/* resource can only be defined at the foreign table level */
+			if (catalog != ForeignTableRelationId)
+			{
+				ereport(ERROR,
+						(errcode(ERRCODE_FDW_INVALID_OPTION_NAME),
+						 errmsg("the resource option can only be defined at the foreign table level")));
+			}
```
Comment (denalex) [inline contrib/pxf_fdw/pxf_option.c]: I'd think at some point we would need to just accumulate the options defined and then do all validation below per catalog object type and extract the core comparison/error reporting into a helper function.

File: contrib/pxf_fdw/expected/pxf_fdw.out [hunk 2]
```diff
@@ -2,24 +2,125 @@
...
+    SERVER dummy_server;
+-- User mapping alteration fails if protocol option is added
+ALTER USER MAPPING FOR pxf_fdw_user
+    SERVER dummy_server
+    OPTIONS ( ADD protocol 'usermappingprotocol' );
+ERROR:  the protocol option can only be defined at the foreign-data wrapper level
+-- User mapping alteration fails if resource option is added
+ALTER USER MAPPING FOR pxf_fdw_user
+    SERVER dummy_server
+    OPTIONS ( resource '/invalid/option/for/user/mapping' );
```
[the remark this answers is not in the document]
Comment (frankgh) [reply contrib/pxf_fdw/expected/pxf_fdw.out]: `ADD` is optional, but will add it to make it consistent

File: contrib/pxf_fdw/expected/pxf_fdw.out [hunk 3]
```diff
@@ -2,24 +2,125 @@
...
+ALTER FOREIGN TABLE dummy_table
+    OPTIONS ( ADD protocol 'table_protocol' );
+ERROR:  the protocol option can only be defined at the foreign-data wrapper level
+-- Table alteration fails if resource option is dropped
+ALTER FOREIGN TABLE dummy_table
+    OPTIONS ( DROP resource );
+ERROR:  the resource option must be defined at the foreign table level
+-- Table alteration fails if resource is provided as an empty string
+ALTER FOREIGN TABLE dummy_table
+    OPTIONS ( SET resource '' );
```
[the remark this answers is not in the document]
Comment (frankgh) [reply contrib/pxf_fdw/expected/pxf_fdw.out]: yeah

File: contrib/pxf_fdw/pxf_option.c (hunk 1 above)
Comment (frankgh) [reply contrib/pxf_fdw/pxf_option.c]: yeah, makes sense

**Pull request merged**